### PR TITLE
P percent rule

### PR DESCRIPTION
**Pull request by MBrouns**
Implements the p_percent fairness metric described in https://arxiv.org/pdf/1507.05259.pdf

Also made a start on implementing the actual classifier they're discussing but that should probably go in a separate PR

### CODE REVIEW

File: sklego/metrics.py [hunk 1]
```diff
@@ -16,3 +16,41 @@ def correlation_metric(estimator, X, y_true=None):
...
+        sensitive_col = X[:, column] if isinstance(X, np.ndarray) else X[column]
+
+        if not np.all((sensitive_col == 0) | (sensitive_col == 1)):
+            raise ValueError(f'neg_p_percent only supports binary indicator columns for `column`. '
+                             f'Found values {np.unique(sensitive_col)}')
+
+        y_hat = estimator.predict(X)
+        y_given_z1 = y_hat[sensitive_col == 1]
+        y_given_z0 = y_hat[sensitive_col == 0]
+        p_y1_z1 = np.sum(y_given_z1 == positive_target) / (len(y_given_z1) or 1)
```
Comment (koaning) [inline sklego/metrics.py]: why not `np.mean` here?

File: tests/test_metrics/test_neg_p_percent.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+def test_neg_p_percent_pandas(sensitive_classification_dataset):
+    X, y = sensitive_classification_dataset
+    mod_unfair = LogisticRegression().fit(X, y)
+    assert neg_p_percent("x2")(mod_unfair, X) == 0
+
+    mod_fair = make_pipeline(
+        ColumnSelector('x1'),
+        LogisticRegression(),
+    ).fit(X, y)
+    assert neg_p_percent("x2")(mod_fair, X) == -0.9
```
Comment (koaning) [inline tests/test_metrics/test_neg_p_percent.py]: just to check if i understand. the score for the fair model is ... worse?

Comment (MBrouns) [reply tests/test_metrics/test_neg_p_percent.py]: yes because lower is better in grid search right?

File: sklego/metrics.py (hunk 1 above)
Comment (MBrouns) [reply sklego/metrics.py]: good one, will change it

File: tests/test_metrics/test_neg_p_percent.py (hunk 2 above)
Comment (koaning) [reply tests/test_metrics/test_neg_p_percent.py]: i thought higher is better? from the [docs](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html) 

image

Comment (MBrouns) [reply tests/test_metrics/test_neg_p_percent.py]: Thanks for catching that one! Inverted the whole thing. Thats what you get for building features without wifi to check the sklearn docs 🙂

Comment (koaning) [reply tests/test_metrics/test_neg_p_percent.py]: Might be one for the `import this` feature. 

> Off by one errors are hard to catch, 
> Flipping the sign can be even harder (and worse).
